gprofiler/profilers/java.py

```diff
@@ -649,6 +668,7 @@ def _profile_process(self, process: Process) -> Optional[StackToSampleCount]:
         with AsyncProfiledProcess(
             process, self._storage_dir, self._buildids, self._mode, self._safemode, self._java_safemode
         ) as ap_proc:
+            self._process_monitor.register_process(process.pid)
```

Comment:
You never remove PIDs from here. @d3dave added some mechanism in his PR: https://github.com/Granulate/gprofiler/pull/219/files#diff-77067d3a05ebeab8e9e2620faefa95849baa469195b8c07aa6283cc160e00df3R98

Well, I guess we can go with your code and whomever merges first can use the other's code. Just make sure your API stands with his.

Reply:
PIDs are removed in the exit callback

Reply:
Yeah I commented that thinking that the removal in the exit callback is incorrect :sweat_smile: